guide/popular-topics/errors.md

```diff
@@ -0,0 +1,227 @@
+
+module.exports = (message, args) =>	{
+	// Should be message.client instead!
+	client.fetchUser(args[0]).then(user => {
+		message.reply('your requested user', user.tag);
+	});
+};
+```
+
+```js
+const { Client } = require('discord.js');
+const client = new Client();
+
+client.on('message', someHandlerFunction);
+
+client.login('token');
+//	client will not be logged in yet!
+client.fetchUser('myId').then(someInitFunction);
+```
+
+<branch version="12.x">
+
+### MessageEmbed field names may not be empty.
+
+This error originates from attempting to call `MessageEmbed.addField()` without the first parameter, which is a title. If you would like the title to be empty for a reason, you should use a zero width space, which can be inputted as `\u200b`.
+
+### MessageEmbed field values may not be empty.
+
+This error, in conjunction to the previous error, is the result of calling `MessageEmbed.addField()` without the second parameter, the value. You can use a zero width space if you would like this empty.
+
+</branch>
+
+### The messages must be an Array, Collection, or number.
+
+This error originates from an invalid call to `bulkDelete()`, make sure you are inputting a valid Array or Collection of messages, or a valid number.
+
+### Members didnt't arrive in time.
+
+Another common error, this error originates from the client requesting members from the API through the websocket, and the member chunks not arriving in time and triggering the timeout. The most common cause to this error is a bad connection, however, it can also be caused by a very large amount of members being fetched, upwards of 50 thousand. To fix this, run the bot on a location with better internet, such as a VPS. If this does not work for you, you will have to manually change the hardcoded member fetching timeout in the source code.
+
+### MaxListenersExceededWarning: Possible EventEmitter memory leak detected...
+
+This error is caused by spawning a large amount of event listeners, usually for the client. The most common cause of this is nesting your event listeners instead of separating them. The way to fix this error is to make sure you do not nest your listeners, it is **not** to use `emitter.setMaxListeners()` as the error suggests.
+
+You can debug these messages in different ways:
+- Through the [CLI](https://nodejs.org/api/cli.html#cli_trace_warnings): `node --trace-warnings index.js`
+- Through the [`process#warning` event](https://nodejs.org/api/process.html#process_event_warning): `process.on('warning', console.warn);`
+
+### Cannot send messages to this user.
+
+This error is thrown when the bot attempts to send a DM message to a user and it is unable to do so. This is caused by a variety of reasons:
+- The bot and the user do not share a guild (oftentimes people accidentally kick or ban a user and then try to dm them).
+- The user has blocked the bot.
+- The user has disabled dms in the privacy settings.
+- The bot is attempting to DM another bot.
```

Comment:
You should move this up since the below text says `In the case of the last two reasons`.

Reply:
woops